src/transformers/generation_utils.py

```diff
@@ -708,6 +717,10 @@ def _get_logits_processor(
             processors.append(ForcedEOSTokenLogitsProcessor(max_length, forced_eos_token_id))
         if remove_invalid_values is True:
             processors.append(InfNanRemoveLogitsProcessor())
+        if length_regulation_factor is not None and length_regulation_start is not None:
```

Comment:
Think it would be cleaner to just have a single input tuple here `exponential_decay_length_penalty`. What do you think @Narsil ?

Reply:
I do agree a tuple is better.

The reason is if a user adds only 1 argument, then the current code would fail (since one would be None and the other have a real value).
So having a tuple forces users to set both at once, making it less likely to crash/fail.

Another thing though, is that this doesn't take into account already existing tokens (either in encode-decoder or in decoder-only). IMHO, this makes this argument harder to use, since you have to offset it with the start_ids, which you might not have access to (for instance pipeline users). `max_length` as the same issue and `max_new_tokens` is there to be agnostic to the initial length. 

If we could make `length_regulation_start` become expressed in generated tokens instead of overall tokens ,it would make a nicer api IMO.

Reply:
Tuple is fine for me. any ideas on an expressive name for the tuple? 

The pipeline arguments makes sense to me, I would change it in the same fashion as max_new_tokens. Do you agree with this change @patrickvonplaten?

Reply:
`config` is allowed to have tuples - tuples are immutable so that's totally fine for configs in my opinion. I'm also to have length regulation more dependent on the number of generated tokens as suggested by @Narsil! `max_new_tokens` is better than `max_length` in this regard. 

Regarding the name, I don't really mind that much as long as its docstring is good and the name closely resembles the logit processor class name.